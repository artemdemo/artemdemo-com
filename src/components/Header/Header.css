.header-wrapper {
  margin-bottom: 20px;
}

.header-container {
  padding: 15px 0;
}

@media (max-width: 576px) {
  .header-container {
    text-align: center;
  }
}

.header-item {
  display: inline-block;
  vertical-align: middle;
}
.header-item__logo {
  width: 50%;
  min-width: 280px;
  font-size: 1.5rem;
  font-weight: 300;
}
@media (max-width: 576px) {
  .header-item__logo {
    width: auto;
    margin-bottom: 10px;
  }
}

.header-item__menu {
  width: 50%;
}
.header-item__menu::after {
  content: '';
  display: table;
  clear: both;
}
@media (max-width: 768px) {
  .header-item__menu {
    width: auto;
  }
}

.header-item-menu {
  float: right;
}

.header-separator {
  border-bottom: 1px solid #dee2e6;
  max-width: 90%;
  margin: 0 auto;
}

.header-logo-link {
  color: initial;
}
.header-logo-link:hover {
  text-decoration: none;
}
