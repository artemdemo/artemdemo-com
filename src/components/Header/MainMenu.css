.main-menu-link:hover {
    text-decoration: none;
}