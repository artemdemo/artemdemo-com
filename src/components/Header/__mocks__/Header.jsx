import React from 'react';

const Header = props => (
    <div data-mock='Header'>
        {props.children}
    </div>
);

export default Header;