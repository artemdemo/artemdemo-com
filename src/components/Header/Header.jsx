import React from 'react';
import PropTypes from 'prop-types';
import {Link} from 'gatsby';
import MainMenu from './MainMenu';
import Container from '../Container/Container';
import {locationProp} from '../Layout/layoutProps';
import UpdateReadyInfo from '../UpdateReadyInfo/UpdateReadyInfo';
import './Header.css';

const Header = (props) => {
  const {location, title} = props;
  return (
    <div className='header-wrapper'>
      <Container>
        <div className='header-container'>
          <div className='header-item header-item__logo'>
            <Link className='header-logo-link' to='/'>
              {title}
            </Link>
          </div>
          <div className='header-item header-item__menu'>
            <div className='header-item-menu'>
              <MainMenu location={location}/>
            </div>
          </div>
        </div>
        <div className='header-separator' />
        <UpdateReadyInfo/>
      </Container>
    </div>
  );
};

Header.propTypes = {
  location: locationProp,
  title: PropTypes.string,
};

Header.defaultProps = {
  location: {},
  title: '',
};

export default Header;
