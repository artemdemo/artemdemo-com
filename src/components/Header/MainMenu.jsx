import React from 'react';
import styled from 'styled-components';
import { Link } from 'gatsby';
import { locationProp } from '../Layout/layoutProps';
import * as styles from '../../services/styles';

import './MainMenu.css';

const MainMenuSty = styled.ul`
    ${styles.clearFix}
    ${styles.resetUl}
`;

const MainMenuItemSty = styled.li`
    float: left;
    padding: 0 10px;
    ${props => props.active && `
        border-bottom: 2px solid #96c9ff;
    `}
`;

const MainMenu = (props) => {
    const { location } = props;
    return (
        <MainMenuSty>
            <MainMenuItemSty active={location.pathname === '/' || location.pathname.startsWith('/blog')}>
                <Link className='main-menu-link' to='/'>
                    Blog
                </Link>
            </MainMenuItemSty>
            <MainMenuItemSty active={location.pathname.startsWith('/about')}>
                <Link className='main-menu-link' to='/about'>
                    About
                </Link>
            </MainMenuItemSty>
            <MainMenuItemSty active={location.pathname.startsWith('/contact')}>
                <Link className='main-menu-link' to='/contact'>
                    Contact me
                </Link>
            </MainMenuItemSty>
        </MainMenuSty>
    );
};

MainMenu.propTypes = {
    location: locationProp,
};

MainMenu.defaultProps = {
    location: {},
};

export default MainMenu;