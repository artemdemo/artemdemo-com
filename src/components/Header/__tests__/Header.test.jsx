import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../Header';

jest.mock('../MainMenu');
jest.mock('../../Container/Container');

describe('Header', () => {
    it('should render empty', () => {
        const tree = renderer.create(
            <Header />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should render with title', () => {
        const tree = renderer.create(
            <Header
                title='Test title'
            />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
