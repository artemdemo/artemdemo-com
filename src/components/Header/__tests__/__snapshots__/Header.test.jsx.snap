// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render empty 1`] = `
<div
  className="Header__HeaderSty-x4ek9t-0 ctjVwy"
>
  <div
    data-mock="Container"
  >
    {}
    <div
      className="Header__HeaderContainer-x4ek9t-1 fzAqQm"
    >
      <div
        className="Header__HeaderItem-x4ek9t-2 jLKXMa"
      >
        <div
          data-mock="Link"
        >
          {
  "to": "/",
  "className": "Header__HeaderLogoLink-x4ek9t-5 ehZhfd"
}
          
        </div>
      </div>
      <div
        className="Header__HeaderItem-x4ek9t-2 hoLYPK"
      >
        <div
          className="Header__HeaderItemMenu-x4ek9t-3 imOZsT"
        >
          <div
            data-mock="MainMenu"
          >
            {
  "location": {}
}
          </div>
        </div>
      </div>
    </div>
    <div
      className="Header__HeaderSeparator-x4ek9t-4 eeJXhx"
    />
  </div>
</div>
`;

exports[`Header should render with title 1`] = `
<div
  className="Header__HeaderSty-x4ek9t-0 ctjVwy"
>
  <div
    data-mock="Container"
  >
    {}
    <div
      className="Header__HeaderContainer-x4ek9t-1 fzAqQm"
    >
      <div
        className="Header__HeaderItem-x4ek9t-2 jLKXMa"
      >
        <div
          data-mock="Link"
        >
          {
  "to": "/",
  "className": "Header__HeaderLogoLink-x4ek9t-5 ehZhfd"
}
          Test title
        </div>
      </div>
      <div
        className="Header__HeaderItem-x4ek9t-2 hoLYPK"
      >
        <div
          className="Header__HeaderItemMenu-x4ek9t-3 imOZsT"
        >
          <div
            data-mock="MainMenu"
          >
            {
  "location": {}
}
          </div>
        </div>
      </div>
    </div>
    <div
      className="Header__HeaderSeparator-x4ek9t-4 eeJXhx"
    />
  </div>
</div>
`;
